src/StatsdClient/Statsd.cs

```diff
@@ -315,6 +340,21 @@ public void Send(string command)
             }
         }
 
+        public async Task SendAsync(string command)
+        {
+            try
+            {
+                await Udp.SendAsync(command).ConfigureAwait(false);
+                // clear buffer (keep existing behavior)
+                if (Commands.Count > 0)
+                    Commands = new List<string>();

```

Comment:
Couldn't this introduce a race condition with `SendAsync()`?
E.g
* `Commands` has exactly 1 element
* `SendAsync()` reads `Commands.Count`
* `SendAsync(string command)` overwrites `Commands` with an empty `List<string>`
* SendAsync() tries to access the empty `Commands` at index `0`.

Reply:
Don't see how it introduces more of a race condition than the original non-async code.

Reply:
I haven't tested this out, but in simplified form.
```
Task a = client.SendAsync();
Task b = client.SendAsync("command");
await a;
await b;
```

This would probably not be a problem for the current sync overloads, as they're executed sequentially.
```
client.Send();
client.Send("command");
```

Reply:
sure.  but isn't client in your example a singleton being executed against multiple threads.

The DogStatsdService for instance contains loads of static methods.

Reply:
The example was purely contrived to illustrate a *potential* issue.
As you mention this could also be hit in the existing sync code, if the two methods are called from separate threads.
So this is not a *new* issue, but with `async` it might be easier to trigger.

A workaround to hide the underlying issue.
```c#
public Task SendAsync()
{
    List<string> commands = Commands;
    int count = commands.Count;
    if (count < 1)
        return Task.CompletedTask;

    return SendAsync(1 == count ? commands[0] : string.Join("\n", commands.ToArray()));
}
```

Reply:
Yep have no issue applying that.  But it really isn't a complete fix either.
This could allow for multiple duplicate messages to be sent

What do the maintainers think @lucaspimentel ?

Reply:
As it is an existing issue, there is no need to change this code in this PR.
I will fix this issue later and before releasing a new nugget package.